client/src/providers/UserProvider.tsx

```diff
@@ -42,10 +47,14 @@ export function UserProvider(props: React.PropsWithChildren<{}>): JSX.Element {
   ): Promise<void> {
```

Comment:
while you're here can you change this terrible code (written by me)
the `user` arg here clashes with the `user` state above
maybe name this one `currentUser`??